src/ut_main.c

```diff
@@ -29,8 +29,10 @@
 #include <ut.h>
 #include <ut_log.h>
 #include <ut_kvp_profile.h>
-#include "ut_internal.h"
+#include <ut_internal.h>
```

Comment:
You shouldn't need to check things like `UT_CUNIT`  because it should really be the same between VARIANTS.

What you need to implement is the wrapper to gest, 

`ut_cunit.c` is the wrapper from ut-main to cunit.

`ut_gtest.c` should be the new implmentation for the wrapper for ut-main to gtest, so it would implement functions like `UT_log_setLogFilePath` which is required.

Reply:
Well, with the current structure I need to, as ut_main.c is used by both C and CPP variants.
However, few of the functions are not used by CPP like, UT_set_results_output_filename(), UT_Manage_Suite_Activation(), UT_Manage_Suite_Activation(), UT_set_test_mode().

The wrappers for gtest are all header files like ut_test_fixture.h  ut_test_runner.h, as they are all classes.

Reply:
your missing the point.

there's no switches required to swap between C&C++

ut-main.c is the same for both, and calls the correct functions today.

you have a wrapper for functions like UT_set_test_mode() that is required to implement on both... You have one for C & another for C++ that implements all functions.

UT_set_results_output_filename()
UT_set_test_mode()


are implemented in both